packages/lit-helpers/src/spread.ts

```diff
@@ -186,68 +184,80 @@ export const spreadEvents = directive(SpreadEventsDirective);
+    for (const key in data) {
+      const value = data[key];
+      if (value === prevData[key]) {
+        continue;
+      }
+      const name = key.slice(1);
+      switch (key[0]) {
+        case '@': // event listener
+          this.eventData[name] = value;
+          this.applyEvent(name, value as EventListenerWithOptions);
+          break;
+        case '.': // property
+          safeSetProperty(element, name, value);
+          break;
+        case '?': // boolean attribute
+          if (value) {
+            element.setAttribute(name, '');
+          } else {
+            element.removeAttribute(name);
+          }
+          break;
+        default:
+          // standard attribute
+          if (value != null) {
+            element.setAttribute(key, String(value));
+          } else {
+            element.removeAttribute(key);
+          }
+          break;
+      }
     }
+  }
 
-    groom(data: { [key: string]: unknown }) {
-        const { prevData, element } = this;
-        if (!prevData) return;
-        for (const key in prevData) {
-            const name = key.slice(1);
-            if (!data || (!(key in data) && element[name] === prevData[key])) {
-                switch (key[0]) {
-                    case '@': // event listener
-                        this.groomEvent(name, prevData[key] as EventListenerWithOptions)
-                        break;
-                    case '.': // property
-                        // @ts-ignore
-                        element[name] = undefined;
-                        break;
-                    case '?': // boolean attribute
-                        element.removeAttribute(name);
-                        break;
-                    default:
-                        // standard attribute
-                        element.removeAttribute(key);
-                        break;
-                }
-            }
+  groom(data: { [key: string]: unknown }) {
+    const { prevData, element } = this;
+    if (!prevData) return;
+    for (const key in prevData) {
+      const name = key.slice(1);
+      if (!data || (!(key in data) && element[name] === prevData[key])) {
+        switch (key[0]) {
+          case '@': // event listener
+            this.groomEvent(name, prevData[key] as EventListenerWithOptions);
+            break;
+          case '.': // property
+            safeSetProperty(element, name, undefined);
+            break;
+          case '?': // boolean attribute
+            element.removeAttribute(name);
+            break;
+          default:
+            // standard attribute
+            element.removeAttribute(key);
+            break;
         }
+      }
     }
+  }
+}
+
+function safeSetProperty(element: Element, name: string, value: unknown) {
+  if (hasSetter(element, name)) {
+    element[name] = value;
+  } else {
+    console.warn(
+      `Could not set property "${name}" on ${element.tagName} because it has no "setter".`,
+    );
+  }
```

Comment:
We probably want to do something like https://github.com/thepassle/app-tools/tree/master/env so that console warnings can be trimmed from the final deployment in a production environment.

Reply:
Perfect! I was thinking the same thing, but wasn't sure if there was already a mechanism in place for this.